pkgs/by-name/bi/bindiff/package.nix

```diff
@@ -0,0 +1,78 @@
+{
+  buildFHSEnv,
+  fetchurl,
+  dpkg,
+  makeWrapper,
+  lib,
+  stdenv,
+  jdk17,
+  fontconfig,
+  makeDesktopItem,
+}:
+let
+  meta = {
+    description = "Quickly find differences and similarities in disassembled code";
+    homepage = "https://zynamics.com/bindiff.html";
+    changelog = "https://github.com/google/bindiff/releases/tag/v8";
+    license = lib.licenses.asl20;
+    platforms = [ "x86_64-linux" ];
+    maintainers = [ lib.maintainers.BonusPlay ];
+  };
+  desktopItem = makeDesktopItem {
+    name = "bindiff";
+    exec = "bindiff";
+    comment = meta.description;
+    desktopName = "BinDiff";
+    genericName = "BinDiff";
+    categories = [ "Development" ];
+    icon = "com-google-security-zynamics-bindiff";
+  };
+  pkg = stdenv.mkDerivation {
+    src = fetchurl {
+      url = "https://github.com/google/bindiff/releases/download/v8/bindiff_8_amd64.deb";
+      sha256 = "sha256-ghmQ45dKnfZzN5Q3DkhUhqwna6XXd6BrTr5XXwkvTdo=";
+    };
+    version = "8";
+    pname = "bindiff";
+
+    nativeBuildInputs = [
+      makeWrapper
+      dpkg
+    ];
+
+    dontBuild = true;
+
+    unpackPhase = ''
+      dpkg-deb -x "$src" ./
+    '';
+
+    installPhase = ''
+      mkdir -p "$out/usr" "$out/opt"
+      cp -a usr/* "$out/usr"
+      cp -a opt/* "$out/opt"
+      cp -a etc/* "$out/etc"
+    '';
+
+    postFixup = ''
+      wrapProgram $out/opt/bindiff/bin/bindiff \
+        --set JAVA_HOME ${jdk17.home} \
+        --chdir /opt/bindiff
+
+      wrapProgram $out/opt/bindiff/bin/bindiff_ui \
+        --set JAVA_HOME ${jdk17.home} \
+        --set FONTCONFIG_FILE ${fontconfig.out}/etc/fonts/fonts.conf \
+        --chdir /opt/bindiff
+    '';
+
+    inherit desktopItem;
+    inherit meta;
+  };
+in
+buildFHSEnv {
+  name = "bindiff";
+  targetPkgs = pkgs: [ pkg ];
+  runScript = "${pkg}/opt/bindiff/bin/bindiff -ui";
```

Comment:
Why is UI flag added here automatically (in contrast to upstream)? I believe `bindiff` in nixos should behave exactly like `bindiff` in upstream, without non-default flags added - especially since there are use-cases for non-ui processing with bindiff

Reply:
You are correct. I had some trouble configuring both at the same time and UI is probably more common usage. I just want to merge something that's working for now, I can follow up with a PR to add headless as well.